google-cloud-spanner/src/test/java/com/google/cloud/spanner/LongRunningSessionsBenchmark.java

```diff
@@ -0,0 +1,317 @@
+@OutputTimeUnit(TimeUnit.SECONDS)
+public class LongRunningSessionsBenchmark {
+  private static final String TEST_PROJECT = "my-project";
+  private static final String TEST_INSTANCE = "my-instance";
+  private static final String TEST_DATABASE = "my-database";
+  private static final int HOLD_SESSION_TIME = 100;
+  private static final int LONG_HOLD_SESSION_TIME = 10000; // 10 seconds
+  private static final int RND_WAIT_TIME_BETWEEN_REQUESTS = 100;
+  private static final Random RND = new Random();
+
+  @State(Scope.Thread)
+  @AuxCounters(org.openjdk.jmh.annotations.AuxCounters.Type.EVENTS)
+  public static class BenchmarkState {
+    private StandardBenchmarkMockServer mockServer;
+    private Spanner spanner;
+    private DatabaseClientImpl client;
+
+    @Param({"100"})
+    int minSessions;
+
+    @Param({"400"})
+    int maxSessions;
+
+    @Param({"4"})
+    int numChannels;
+
+    /** AuxCounter for number of RPCs. */
+    public int numBatchCreateSessionsRpcs() {
+      return mockServer.countRequests(BatchCreateSessionsRequest.class);
+    }
+
+    /** AuxCounter for number of sessions created. */
+    public int sessionsCreated() {
+      return mockServer.getMockSpanner().numSessionsCreated();
+    }
+
+    @Setup(Level.Invocation)
+    public void setup() throws Exception {
+      mockServer = new StandardBenchmarkMockServer();
+      TransportChannelProvider channelProvider = mockServer.start();
+
+      /**
+       * This ensures that the background thread responsible for cleaning long-running sessions
+       * executes every 10s. Any transaction for which session has not been used for more than 2s
+       * will be treated as long-running.
+       */
+      InactiveTransactionRemovalOptions inactiveTransactionRemovalOptions =
+          InactiveTransactionRemovalOptions.newBuilder()
+              .setActionOnInactiveTransaction(ActionOnInactiveTransaction.WARN_AND_CLOSE)
+              .setExecutionFrequency(Duration.ofSeconds(10))
+              .setIdleTimeThreshold(Duration.ofSeconds(2))
+              .build();
+      SpannerOptions options =
+          SpannerOptions.newBuilder()
+              .setProjectId(TEST_PROJECT)
+              .setChannelProvider(channelProvider)
+              .setNumChannels(numChannels)
+              .setCredentials(NoCredentials.getInstance())
+              .setSessionPoolOption(
+                  SessionPoolOptions.newBuilder()
+                      .setMinSessions(minSessions)
+                      .setMaxSessions(maxSessions)
+                      .setInactiveTransactionRemovalOptions(inactiveTransactionRemovalOptions)
+                      .build())
+              .build();
+
+      spanner = options.getService();
+      client =
+          (DatabaseClientImpl)
+              spanner.getDatabaseClient(DatabaseId.of(TEST_PROJECT, TEST_INSTANCE, TEST_DATABASE));
+      // Wait until the session pool has initialized.
+      while (client.pool.getNumberOfSessionsInPool()
```

Comment:
nit: you can use `SessionPoolOptions#setWaitForMinSessions(Duration waitForMinSessions)` for this.

(I guess this is a copy-paste from one of the other benchmarks. The above-mentioned method was added later.)

Reply:
Yes, I have done this.